k2/csrc/fsa_algo_test.cc

```diff
@@ -304,44 +300,41 @@ TEST(FsaAlgo, Intersect) {
   }
 
   {
-    std::vector<Arc> arcs_a = {{0, 1, 1}, {1, 2, 0}, {1, 3, 1},
-                               {1, 4, 2}, {2, 2, 1}, {2, 3, 1},
-                               {2, 3, 2}, {3, 3, 0}, {3, 4, 1}};
-    std::vector<int32_t> arc_indexes_a = {0, 1, 4, 7, 9};
-    Fsa a;
-    a.arc_indexes = std::move(arc_indexes_a);
-    a.arcs = std::move(arcs_a);
-
-    std::vector<Arc> arcs_b = {
-        {0, 1, 1},
-        {1, 3, 1},
-        {1, 2, 2},
-        {2, 3, 1},
-    };
-    std::vector<int32_t> arc_indexes_b = {
-        0,
-        1,
-        3,
-        4,
-    };
-    Fsa b;
-    b.arc_indexes = std::move(arc_indexes_b);
-    b.arcs = std::move(arcs_b);
+    std::vector<Arc> arcs_a = {{0, 1, 1},
+                               {1, 2, 0},
+                               {1, 3, 1},
+                               {1, 4, 2},
+                               {2, 2, 1},
+                               {2, 3, 1},
+                               {2, 3, 2},
+                               {3, 3, 0},
+                               {3, 4, 1}};
+    Fsa a(std::move(arcs_a), 4);
```

Comment:
the final state 4 can be inferred from `arcs_a`.

Reply:
See above (Fsa with all states are not co-accessible)